crates/node/src/lib.rs

```diff
@@ -356,18 +357,18 @@ async fn handle_line(node: &mut Node, line: String) -> eyre::Result<()> {
                             let mut iter = args.split(' ');
                             let type_ = iter.next()?;
                             let resource = iter.next()?;
-                            let version = iter.next();
-                            let metadata = iter.next()?.as_bytes().to_vec();
+                            let version = iter.next()?;
+                            let metadata = iter.next();
 
                             Some((type_, resource, version, metadata))
                         }) else {
                             println!(
-                                "{IND} Usage: application install <\"url\"|\"file\"> <resource> [version] <metadata>"
+                                "{IND} Usage: application install <\"url\"|\"file\"> <resource> [version] [metadata]"
                             );
                             break 'done;
                         };
 
-                        let Ok(version) = version.map(|v| v.parse()).transpose() else {
+                        let Ok(version) = version.parse::<Version>() else {
```

Comment:
you can avoid the import up top if this isn't explicitly defined, rust should be able to infer it properly from `install_application_from{url,path}`